.github/workflows/main.yaml

```diff
@@ -7,8 +7,13 @@ name: Build and test
 on:
   push:
     branches: [ main ]
+
   pull_request:
-    branches: [ main ]
+      types:
+          - opened
+          - reopened
+          - synchronize
+          - ready_for_review
```

Comment:
run linting and test on all PRs in a non-draft state, regardless of what the base branch is